src/formnetworks/bilinearformnetwork.jl

```diff
@@ -60,9 +60,16 @@ function BilinearFormNetwork(
 )
   @assert issetequal(flatten_siteinds(bra), flatten_siteinds(ket))
   link_space = isempty(flatten_siteinds(bra)) ? 1 : nothing
-  operator_inds = union_all_inds(siteinds(ket), dual_site_index_map(siteinds(ket)))
-  # TODO: Define and use `identity_network` here.
-  O = ITensorNetwork(Op("I"), operator_inds; link_space)
+  s = siteinds(ket)
+  s_mapped = dual_site_index_map(s)
+  operator_inds = union_all_inds(s, s_mapped)
+  constructor_f =
+    v -> inds -> if !isempty(inds)
+      reduce(*, [delta(dag(s), sm) for (s, sm) in zip(s[v], s_mapped[v])])
+    else
+      ITensor(one(Bool))
+    end
+  O = ITensorNetwork(constructor_f, operator_inds; link_space)
```

Comment:
```julia
function itensor_identity_map(i_pairs::Vector{<:Pair})
  return prod(i_pairs; init=ITensor(one(Bool))) do i_pair
    return delta(Bool, dag(first(i_pair)), last(i_pair))
  end
end
```
and then:
```suggestion
  O = ITensorNetwork(operator_inds; link_space) do v
    # Use `itensor_identity_map` on `s .=> s_mapped`.
  end
```
Hopefully that makes sense. There is something a bit funny that `operator_inds` is being passed but not really being used.

Reply:
I changed it to that. Although I had to lower the type restriction on `itensor_identity_map` to avoid it erroring in the case `s[v] = Index[]` and `s_mapped[v] = Index[]`.